android/source/VideoLocker/src/org/edx/mobile/view/custom/WebClient.java

```diff
@@ -0,0 +1,105 @@
+package org.edx.mobile.view.custom;
+
+import android.app.Activity;
+import android.webkit.WebChromeClient;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import org.edx.mobile.logger.Logger;
+import org.edx.mobile.util.BrowserUtil;
+
+import java.lang.ref.WeakReference;
+
+/**
+ * Created by rohan on 2/2/15.
+ *
+ * This class represents a custom {@link android.webkit.WebViewClient}.
+ * This class is responsible for setting up a given {@link android.webkit.WebView}, assign itself
+ * as a {@link android.webkit.WebViewClient} delegate and to intercept URLs being loaded.
+ * Depending on the form of URL, this client may forward URL back to the app.
+ */
+public class WebClient extends WebViewClient {
+
+    private Logger logger = new Logger(WebClient.class);
+    private WeakReference<Activity> activity;
+
+    public WebClient(Activity activity, WebView webView) {
+        this.activity = new WeakReference<Activity>(activity);
+        setupWebView(webView);
```

Comment:
Absolutely, I changed the code and now we don't need to hold onto an activity reference.